javalib/src/main/scala/java/lang/process/GenericProcess.scala

```diff
@@ -1,7 +1,163 @@
 package java.lang.process
 
 import scala.scalanative.unsafe._
+import scala.scalanative.meta.LinktimeInfo
+
+import java.time.{Duration, Instant}
+import java.{util => ju}
+import ju.{Optional, Arrays}
+import ju.concurrent.{CompletableFuture, ConcurrentHashMap, TimeUnit}
+
+abstract class GenericProcess() extends java.lang.Process {
+  private lazy val handle = new GenericProcess.Handle(this)
 
-abstract class GenericProcess extends java.lang.Process {
   private[lang] def checkResult(): CInt
+  private[process] def processInfo: GenericProcess.Info
+
+  override def toHandle(): ProcessHandle = handle
+
+  override def onExit(): CompletableFuture[Process] =
+    toHandle().onExit().thenApply(_ => this)
+
+  override def info(): ProcessHandle.Info = processInfo
+  override def pid(): scala.Long = processInfo.pid
+}
+
+private object GenericProcess {
+  // Represents ProcessHandle for process started by Scala Native runtime
+  // Cannot be used with processes started by other programs
+  class Handle(val process: GenericProcess) extends ProcessHandle {
+    override def parent(): Optional[ProcessHandle] = Optional.empty()
+
+    // We don't track transitive children
```

Comment:
To me the "sense" here is strange. By my reading of the JDK 23
`ProcessHandle` doc, the `descendents()` are the children and
the transitive children.

Perhaps something more direct, with less mental tracing, such as:
```
// We track neither children nor transitive children
   override def children(): ju.stream.Stream[ProcessHandle] =
      ju.stream.Stream.empty()
    override def descendants(): ju.stream.Stream[ProcessHandle] =   
      ju.stream.Stream.empty()
```